tests/test_build/source/code.js

```diff
@@ -149,3 +149,9 @@ const ExampleAttribute = null;
  * @param {String} p2.bar
  */
 function destructuredParams(p1, {foo, bar}) {}
+
+/**
+ * @param {number} [p1=42]
```

Comment:
If possible, it would be nice to test one of these params with the default specified only in the JSDoc comment and the other with it only in the function definition. That way, we will notice if we (or jsdoc) ever changes its behavior.